src/licenses.ts

```diff
@@ -0,0 +1,39 @@
+import {Change, ChangeSchema} from './schemas'
+
+/**
+ * Loops through a list of changes, filtering and returning the
+ * ones that don't conform to the licenses allow/deny lists.
+ * @param {Change[]} changes The list of changes to filter.
+ * @param { { allow?: string[], deny?: string[]}} licenses An object with `allow`/`deny` keys, each containing a list of licenses.
+ * @returns {Array<Change} The list of denied changes.
+ */
+export function getDeniedLicenseChanges(
+  changes: Array<Change>,
+  licenses: {
+    allow?: Array<string>
+    deny?: Array<string>
+  }
+): Array<Change> {
+  let {allow = null, deny = null} = licenses
+
+  let disallowed: Change[] = []
+
+  for (const change of changes) {
+    let license = change.license
+    // TODO: be loud about unknown licenses
+    if (license === null) {
+      continue
+    }
+    if (allow !== null) {
```

Comment:
This condition will never *not* be true. 

For optional object parameters like `allow?: Array<string>`, the value will be an array of string (if defined) or `undefined` (if not defined).

In JS, `undefined` is not strictly equal to `null` (`undefined !== null` is true). However, `undefined` is loosely equal to `null` (`undefined != null` is false). 

Beware that an empty array is never equal to null, either with strict or loose equality. So passing an empty array would also evaluate this to true.

Might be worth a few more unit tests to make sure this is correct.

Reply:
The original intent was to know if we were dealing with default parameters or if the user was actually providing a list. @hmaurer called out this point yesterday too, and I made some changes in cc22dcd. What do you think? 

[These two tests](https://github.com/actions/dependency-review-action/blob/adding-lists/__tests__/licenses.test.ts#L49-L59) now exercise this code path.

Reply:
I think this addresses the major concern. Passing an empty list will still be true for `allow !== undefined`, which may lead to surprising behaviors if someone were to call this function with an empty list, but a lesser concern.